builder.md

```diff
@@ -1,57 +1,142 @@
+example: [the solarized scheme](https://github.com/base16-project/base16-schemes/blob/main/solarized-dark.yaml)
 
-- `/*.yaml`
+Builders are designed for lower-level ("plumbing") usage, specifically for
+scripting and as component to build more complex base16 applications.
 
-### Template Repository
+The more complex apps designed for end users ("porcelain") are usually referred
+to as **managers**, and they don't need to follow any standard besides helping
+with base16 theming somehow, in a way that makes sense for their intended
+usecase.
 
-Each template repository should have a templates folder containing a config.yaml and any needed mustache template files.
+Template maintainers SHOULD provide built versions (with all existing scheme)
+so the end user doesn't need to be aware of the builder.
 
-- `/templates/*.mustache` - A template file (there may be multiples of these)
-- `/templates/config.yaml` - A template configuration file
+## Interface
 
-## Workflow
-The first thing a builder needs to do is parse all the scheme files from the schemes repository (as defined in the [file guidelines](https://github.com/chriskempson/base16/blob/master/file.md)). All files matching the pattern `*.yaml` should be loaded from the root of the schemes repository.
+All base16 builders MUST provide a single feature: building a template using 1 or more schemes.
 
-When building a target template repository, a base16 builder should first clear out any old output. Next, for all templates defined in the template repo's config file (located at `/templates/config.yaml`), the builder should iterate through all the defined schemes and output matching files. The built filename should look like `[output-dir]/base16-[slug][extension]`, where the slug is taken from the scheme filename made lowercase with spaces replaced with dashes and both the extension and output-dir are taken from `/template/config.yaml`.
+### CLI
 
-In the case where schemes share the same slug, a builder will overwrite files previously generated from the template. Should this happen, a builder should show warning messages listing the overwritten files.
+It is REQUIRED that that this functionality is exposed as binary CLI executable
+with the exact interface described below.
+
+The binary name SHOULD contain `base16`, but is otherwise left as a choice to
+the author.
+
+To be fully compliant, a builder CLI interface MUST NOT include any other
+feature, option, argument, or deviance from the expected interface and
+behaviour of the program.
+
+<!-- TODO: For convenience, we make a manpage and set of tests available. All
+compliant builders MUST fully conform to these two. -->
+
+The CLI interface MUST work without relying on any network connection.
+
+```bash
+base16 TEMPLATES-DIRECTORY SCHEME-FILE ...
+```
+
+`TEMPLATES-DIRETORY` being a directory containing a `config.yaml`, as well as at
+least one `.mustache` template. These are usually named `templates` on all
```

Comment:
Is this `TEMPLATES-DIRECTORY` the `templates` directory within a template repo or the template repo itself? i.e. https://github.com/base16-project/base16-vim or https://github.com/base16-project/base16-vim/tree/main/templates?

Edit: from reading more in the "Output and behaviour" below, it seems this is the template dir (or whatever directory contains the `.mustache` and `.config.yaml` files and the `config.yaml` used a relative path to point to the output dir.

Reply:
Yup! That's right. The idea is to make it as easy as possible for CI usage (on most repos it is simply `templates`), while allowing for more advanced usage (such as keeping a set of templates somewhere and building them).